packages/cli/src/commands/build/index.ts

```diff
@@ -807,3 +811,20 @@ function mergeFlags(
     return [];
   });
 }
+
+function isPackageInstalled(name: string, pkg: PackageJson | null): boolean {
+  const dependencies = [
+    ...Object.keys(pkg?.dependencies ?? {}),
+    ...Object.keys(pkg?.devDependencies ?? {}),
+  ];
```

Comment:
Right now these packages are iterated twice because `isPackageInstalled()` is always called twice. 

Let's avoid that by hoisting out of this function. 

You could even use a Set so that you only iterate once and lookups are much faster.

Reply:
For example:

```js
const dependencies = new Set();
for (const key in pkg?.dependencies ?? {}) {
  dependencies.add(key);
}
for (const key in pkg?.devDependencies ?? {}) {
  dependencies.add(key);
}


if (dependencies.has('@vercel/speed-insights')) {
  // do stuff
}

if (dependencies.has('@vercel/analytics')) {
  // do stuff
}
```

Reply:
Good call. @chriswdmr suggested that we convey the package versions instead of a boolean.
I made it a map, and it's built only once

Reply:
In the unlikely event of the same package being in both `devDependencies` and `dependencies`, the later will take over.